tests/unit/test_charm.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+#
+# Learn more about testing at: https://juju.is/docs/sdk/testing
+
+import unittest
+
+from ops import testing
+from ops.model import ActiveStatus, BlockedStatus
+from ops.testing import Harness
+
+from charm import Route53AcmeOperatorCharm
+
+testing.SIMULATE_CAN_CONNECT = True
+
+
+class TestCharm(unittest.TestCase):
+    def setUp(self):
+        self.harness = Harness(Route53AcmeOperatorCharm)
+        self.harness.set_leader(True)
+        self.harness.set_can_connect("lego", True)
+        self.addCleanup(self.harness.cleanup)
+        self.harness.begin()
+        self.r_id = self.harness.add_relation("certificates", "remote")
+        self.harness.add_relation_unit(self.r_id, "remote/0")
+
+    def test_given_config_changed_when_email_is_valid_then_status_is_active(self):
```

Comment:
The given and when are inverted. The `when` is the actual action, in this case, `config_changed`. So it should be `given_email_is_valid_when_config_changed_then_status_is_active`. The same issue is present in other tests.

Reply:
✅